### DISCUSSION TITLE
running two stepper motors simultaneously

### DISCUSSION THREAD

**Discussion by loke-bager** [MrYsLab/telemetrix#44] [Ideas]
This is a really useful library. I am able to connect two motors, and first move the first motor, and then the second motor. However I would also like to be able to move both simultaneously. Is this possible in any way? It would be nice with some examples with multiple stepper motors.

**Comment by MrYsLab** [comment]
Hi @loke-bager,
Stepper support utilizes the [AccelStepper library](https://github.com/waspinator/AccelStepper) in the Telemtrix4Arduino sketch. AccelStepper does have a class to run multiple steppers, but if you look at the [code](https://github.com/waspinator/AccelStepper/blob/a9bf2f93bffff950469485c8a91c6e8709bba444/src/MultiStepper.cpp#L52), it runs the motors sequentially and not simultaneously. 
When writing telemetrix, I attempted to use the multiStepper class, but it did not function correctly, so I did not include support for it.

**Comment by MrYsLab** [comment]
@loke-bager I found this youtube video demonstrating running two steppers simultaneously. Essentially, the motors alternate running 1 step each.
The sketch code is included on the video page. To get a similar scheme to work with telemetrix, you would need to port the control logic to Python and then use telemetrix to control the motors. No changes would be made on the Arduino side. 
Hopefully, this may help you accomplish your task.
